doubleml/double_ml_irm.py

```diff
@@ -140,6 +146,150 @@ def __init__(self,
         self.trimming_rule = trimming_rule
         self.trimming_threshold = trimming_threshold
 
+    def cate(self, cate_var: str, method: str, alpha: float, n_grid_nodes: int,
+             n_samples_bootstrap: int, cv: bool, poly_degree: int = None, splines_knots: int = None,
+             splines_degree: int = None, ortho: bool = False, x_grid: np.array = None) -> dict:
```

Comment:
In the code base, we so far don't work with type hints. Such a change of code style should not come together with a new feature. Therefore, please remove them.

```suggestion
    def cate(self, cate_var, method, alpha, n_grid_nodes,
             n_samples_bootstrap, cv, poly_degree=None, splines_knots=None,
             splines_degree=None, ortho=False, x_grid=None)t:
```

General remark on type hints: Adding type hints is a bigger decision and would also require numerous adaptions in contributing guidelines, etc. Some of the larger python packages encourage to use type hints ((PEP 484)[https://peps.python.org/pep-0484/]), see for example `pandas`. Others not yet, see e.g. the discussions https://github.com/scikit-learn/scikit-learn/issues/16705 & https://github.com/fairlearn/fairlearn/pull/592.